src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+using Newtonsoft.Json;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.Tools.Common;
+using Project = Microsoft.DotNet.ProjectModel.Project;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public class MigrateProjectDependenciesRule : IMigrationRule
+    {
+        private readonly ITransformApplicator _transformApplicator;
+        private string _projectDirectory;
+
+        public MigrateProjectDependenciesRule(TransformApplicator transformApplicator = null)
+        {
+            _transformApplicator = transformApplicator ?? new TransformApplicator();
+        }
+
+        public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migrationRuleInputs)
+        {
+            _projectDirectory = migrationSettings.ProjectDirectory;
+
+            var csproj = migrationRuleInputs.OutputMSBuildProject;
+            var projectContext = migrationRuleInputs.DefaultProjectContext;
+            var projectExports = projectContext.CreateExporter("_").GetDependencies(LibraryType.Project);
+
+            var projectDependencyTransformResults =
+                projectExports.Select(projectExport => ProjectDependencyTransform.Transform(projectExport));
+            var propertyTransformResults = new []
+            {
+                AutoUnifyTransform.Transform(true),
+                DesignTimeAutoUnifyTransform.Transform(true)
+            };
+
+            if (projectDependencyTransformResults.Any())
```

Comment:
If you are only going to transfor the propertyTransformResults if there are any project dependencies, why not move its declaration inside the if block?